adapta/utils/decorators/_rate_limit.py

```diff
@@ -0,0 +1,47 @@
+"""
+Rate limit decorator.
+"""
+from functools import wraps
+from typing import Callable
+
+from limits.storage import storage_from_string
+from limits.strategies import STRATEGIES
+from limits.util import parse
+
+from adapta.utils._common import doze
+
+
+def _default_delay_func() -> None:
+    doze(1)
+
+
+def rate_limit(
+    _func: Callable = None,
+    *,
+    limit: str,
+    strategy: str = "moving-window",
+    delay_func: Callable[[], int] = _default_delay_func
+) -> Callable:
+    """
+    Rate limit decorator.
+    :param limit: the limit string to parse
+    :param strategy: the strategy to use
```

Comment:
Can we have an enum here?

Reply:
No, the caller code should use the same names as the `limits` library uses. If we define our own enum, we may have a compatibility issues.

Reply:
Discussed by voice: will do an enum.